packages/babel-core/src/transformation/file/index.js

```diff
@@ -99,7 +100,19 @@ export default class File extends Store {
     this.code    = "";
     this.shebang = "";
 
-    this.hub = new Hub(this);
+    this.hub = {
+      mark: (type: string, message: string, loc: Object) => {
+        this.metadata.marked.push({
+          type,
+          message,
+          loc
+        });
+      },
+      addHelper: this.addHelper.bind(this),
+      getCode: () => this.code,
+      getScope: () => this.scope,
+      buildError: this.buildCodeFrameError.bind(this)
+    };
```

Comment:
I thought you gonna replace the object with the `Hub` class in `packages/babel-traverse/src/hub.js`. Can we replace it?

Like:
```js
hub: HubInterface;

this.hub = new Hub(this);
```